### PR TITLE
Update the logic in evalEER to be mathematically correct for cases wi…

### PR DESCRIPTION
**Pull request by JStehouwer**
…th few sample or separated classes.

### CODE REVIEW

File: openbr/core/eval.cpp [hunk 1]
```diff
@@ -1286,85 +1291,106 @@ void EvalEER(const QString &predictedXML, QString gt_property, QString distribut
         if (gtLabel == 1) {
             classOneTemplateCount++;
             classOneScores.append(templateScore);
+            if (minClassOneScore > templateScore) minClassOneScore = templateScore;
```
Comment (Keyurpatel93) [inline openbr/core/eval.cpp:1294]: Put ` minClassOneScore = templateScore;` on a new line. Same for the others below to keep code styling consistent

File: openbr/core/eval.cpp [hunk 2]
```diff
@@ -1286,85 +1291,106 @@ void EvalEER(const QString &predictedXML, QString gt_property, QString distribut
...
 
     printf("\n==========================================================\n");
     printf("Class 0 Templates: %d\tClass 1 Templates: %d\tTotal Templates: %d\n",
            classZeroTemplateCount, classOneTemplateCount, classZeroTemplateCount + classOneTemplateCount);
     printf("----------------------------------------------------------\n");
     foreach (float FAR, QList<float>() << 0.2 << 0.1 << 0.05 << 0.01 << 0.001 << 0.0001) {
         const OperatingPoint op = getOperatingPoint(operatingPoints, "FAR", FAR);
-        printf("TAR = %.3f @ FAR = %.4f | Threshold= %.3f\n", op.TAR, FAR, op.score);
+        if (op.score < MAX(minClassOneScore, minClassZeroScore) || op.score > MIN(maxClassOneScore, maxClassZeroScore))
+            printf("TAR =  N/A  @ FAR = %.4f | Threshold= %.3f\n", FAR, MAX(op.score, minClassZeroScore));
```
Comment (Keyurpatel93) [inline openbr/core/eval.cpp:1349]: TAR = 1?

File: openbr/core/eval.cpp [hunk 3]
```diff
@@ -1286,85 +1291,106 @@ void EvalEER(const QString &predictedXML, QString gt_property, QString distribut
...
+            printf("TAR = %.3f @ FAR = %.4f | Threshold= %.3f\n", op.TAR, FAR, op.score);
 
     }
     printf("----------------------------------------------------------\n");
-    foreach (float TAR, QList<float>() << 0.8 << 0.85 << 0.9 << 0.95 << 0.98) {
+    foreach (float TAR, QList<float>() << 0.8 << 0.85 << 0.9 << 0.95 << 0.98 << 0.99) {
         const OperatingPoint op = getOperatingPoint(operatingPoints, "TAR", TAR);
-        printf("FAR = %.3f @ TAR = %.4f | Threshold= %.3f\n", op.FAR, TAR, op.score);
+        if (op.score < MAX(minClassOneScore, minClassZeroScore) || op.score > MIN(maxClassOneScore, maxClassZeroScore))
+            printf("FAR =  N/A  @ TAR = %.4f | Threshold= %.3f\n", TAR, op.score);
```
Comment (Keyurpatel93) [inline openbr/core/eval.cpp:1358]: FAR = 0?

File: openbr/core/eval.cpp [hunk 4]
```diff
@@ -1286,85 +1291,106 @@ void EvalEER(const QString &predictedXML, QString gt_property, QString distribut
...
+        form.append(QString("|  :---:  "));
+        frr.append(score > maxClassOneScore ? QString("|         ") : QString("|  %1  ").arg(QString::number(std::abs(1.0 - op.TAR), 'f', 3)));
+        far.append(score < minClassZeroScore ? QString("|         ") : QString("|  %1  ").arg(QString::number(std::abs(op.FAR), 'f', 3)));
+    }
+
+    printf("\nClass 0 Templates: %d, Class 1 Templates: %d\nClass 0 Range: [%.3f,%.3f], Class 1 Range: [%.3f,%.3f]\n\n",
+        classZeroTemplateCount, classOneTemplateCount, minClassZeroScore, maxClassZeroScore, minClassOneScore, maxClassOneScore);
+    printf("| Threshold %s|\n", qPrintable(thresh));
+    printf("|   :---:   %s|\n", qPrintable(form));
+    printf("|  Genuine  %s|\n", qPrintable(frr));
```
Comment (Keyurpatel93) [inline openbr/core/eval.cpp:1387]: Pring out Genuine (FRR) and Spoof (FAR) for clarity sake for the tables

Comment (JStehouwer) [reply openbr/core/eval.cpp]: Resolved in c437d14.

File: openbr/core/eval.cpp (hunk 3 above)
Comment (JStehouwer) [reply openbr/core/eval.cpp]: Resolved in c437d14.

File: openbr/core/eval.cpp (hunk 2 above)
Comment (JStehouwer) [reply openbr/core/eval.cpp]: Resolved in c437d14.

File: openbr/core/eval.cpp (hunk 1 above)
Comment (JStehouwer) [reply openbr/core/eval.cpp]: Resolved in c437d14.

**Pull request merged**